python/paddle/fluid/tests/unittests/dygraph_to_static/test_transformer.py

```diff
@@ -218,18 +218,144 @@ def train_dygraph(args, batch_generator):
+
+        # define input and reader
+        input_field_names = util.encoder_data_input_fields + util.fast_decoder_data_input_fields
+        input_descs = util.get_input_descs(args, 'test')
+        input_slots = [{
+            "name": name,
+            "shape": input_descs[name][0],
+            "dtype": input_descs[name][1]
+        } for name in input_field_names]
+
+        input_field = util.InputField(input_slots)
+        feed_list = input_field.feed_list
+        loader = fluid.io.DataLoader.from_generator(
+            feed_list=feed_list, capacity=10)
+
+        # define model
+        transformer = Transformer(
+            args.src_vocab_size, args.trg_vocab_size, args.max_length + 1,
+            args.n_layer, args.n_head, args.d_key, args.d_value, args.d_model,
+            args.d_inner_hid, args.prepostprocess_dropout,
+            args.attention_dropout, args.relu_dropout, args.preprocess_cmd,
+            args.postprocess_cmd, args.weight_sharing, args.bos_idx,
+            args.eos_idx)
+
+        out_ids, out_scores = transformer.beam_search(
+            *feed_list,
+            bos_id=args.bos_idx,
+            eos_id=args.eos_idx,
+            beam_size=args.beam_size,
+            max_len=args.max_out_len,
+            batch_size=args.batch_size)
+
+    # This is used here to set dropout to the test mode.
+    test_prog = test_prog.clone(for_test=True)
+
+    # define the executor and program for training
+    exe = fluid.Executor(place)
+
+    util.load(test_prog,
+              os.path.join(args.save_static_model_path, "transformer"), exe)
+
+    loader.set_batch_generator(batch_generator, places=place)
+
+    step_idx = 0
+    for feed_dict in loader:
+        seq_ids, seq_scores = exe.run(
+            test_prog,
+            feed=feed_dict,
+            fetch_list=[out_ids.name, out_scores.name],
+            return_numpy=True)
+        step_idx += 1
+        if step_idx == 10:
+            break
+    return seq_ids
+
+
 class TestTransformer(unittest.TestCase):
     def prepare(self, mode='train'):
         args = util.ModelHyperParams()
         batch_generator = util.get_feed_data_reader(args, mode)
         return args, batch_generator
 
-    def test_train(self):
+    def _test_train(self):
         args, batch_generator = self.prepare(mode='train')
         static_avg_loss = train_static(args, batch_generator)
         dygraph_avg_loss = train_dygraph(args, batch_generator)
         self.assertTrue(np.allclose(static_avg_loss, dygraph_avg_loss))
 
+    def _test_predict(self):
+        args, batch_generator = self.prepare(mode='test')
+        static_res = predict_static(args, batch_generator)
+        dygraph_res = predict_dygraph(args, batch_generator)
+        self.assertTrue(
+            np.allclose(static_res, dygraph_res),
+            msg="static_res: {} \n dygraph_res: {}".format(static_res,
+                                                           dygraph_res))
+
+    def test_check_result(self):
```

Comment:
Just a question, why do we combine _test_train and _test_predict together?

Reply:
Because the trained model will be loaded in prediction, run `_test_predict` after `_test_train`.

Reply:
OK, thank you very much for explanation. LGTM